### PR TITLE
Feat: Controller Functionality

### PR DESCRIPTION
**Pull request by tssala23**

### CODE REVIEW

File: src/app.ts
Comment (tumido) [inline src/app.ts:221]: We want the properly formatted data to come directly from the issue template. No more variable magic should be needed here. Instead you should adjust the issue template to return you "moc/smaug" string directly instead of "smaug" (similarly for all other clusters)

File: src/app.ts
Comment (tumido) [inline src/app.ts:238]: Feel free to adjust the issue template keys, so you don't have to do these mappings here and the data already comes in the right format from the issue. The more you hardcode into the controller the more complex future maintenance and changes gets.

**Review by tumido — CHANGES_REQUESTED** [review]
Copy paseted review comments from here https://github.com/operate-first/robozome/pull/10#pullrequestreview-1038630578 since I didn't know that #10 depends on #9.

File: manifests/base/tasks/run.yaml
Comment (tumido) [inline manifests/base/tasks/run.yaml:1]: How is this task different from #10 and why is it part of this PR (since #10 is about tekton, not this one?)

File: src/app.ts
Comment (tumido) [inline src/app.ts:208]: What is this `if` check used for?

Comment (tssala23) [reply src/app.ts:208]: To check if the template used was the one for onboarding.

File: manifests/base/tasks/run.yaml
Comment (tssala23) [reply manifests/base/tasks/run.yaml]: Accidentally added file to the wrong branch.

File: src/app.ts
Comment (larsks) [inline src/app.ts:214]: Shouldn't the second parameter here be `context` (and `payload` the third parameter).

File: .env
Comment (larsks) [inline .env:1]: This file looks like a local development artifact and should probably be removed from the pull request.

**Review by larsks — COMMENTED** [review]
I think we're just about ready to merge this PR, but I'd like to suggest some things to keep in mind when you're working on code in the future:

A pull request should be focused on a specific set of changes. If this pull request is meant to implement the controller functionality, then it should *also* be setting up git workflows, updating OWNERS, CONTRIBUTORS, etc, or configuring github workflows. @tumido I'm looking at you, because it looks like you contributed to this PR :).

When you bundle a bunch of unrelated changes into a pull request, you end up complicating your life: if there is a problem in, say, the github workflow, it's going to prevent your application code from getting merged. If you keep things separate, code can get merged quickly when its ready, rather than being blocked by unrelated changes.

I *would* like to see the `.env` file removed before you merge this pull request. That looks like something that should just be a local development artifact.

**Review by larsks — APPROVED** [review]

**Comment by larsks** [comment]
@tssala23  this just needs a few really minor formatting fixes for the tests to pass; if you run `pre-commit` locally it will both flag *and correct* the outstanding problems.

**Pull request merged**
